<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Log 2025</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Research Progress</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Progress
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="2025.html">2025</a>
    </li>
    <li>
      <a href="simulation.html">simulation</a>
    </li>
    <li>
      <a href="template.html">wflow_template</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hmutanqilong/PostdocLog">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Log 2025</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>qltan_postdoc_note/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250707code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20250707)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250707code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250707)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhmutanqilongPostdocLogtree426f8e67fb999fa5ccf869ba56631deb3cef276atargetblank426f8e6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hmutanqilong/PostdocLog/tree/426f8e67fb999fa5ccf869ba56631deb3cef276a" target="_blank">426f8e6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhmutanqilongPostdocLogtree426f8e67fb999fa5ccf869ba56631deb3cef276atargetblank426f8e6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/hmutanqilong/PostdocLog/tree/426f8e67fb999fa5ccf869ba56631deb3cef276a" target="_blank">426f8e6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/2025.Rmd</code>) and HTML (<code>docs/2025.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/426f8e67fb999fa5ccf869ba56631deb3cef276a/analysis/2025.Rmd" target="_blank">426f8e6</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-10-23
</td>
<td>
Update_2025_10_21
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/426f8e67fb999fa5ccf869ba56631deb3cef276a/docs/2025.html" target="_blank">426f8e6</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-10-23
</td>
<td>
Update_2025_10_21
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/d69197f4d52e66c491e4870762af6fbe883cbe13/docs/2025.html" target="_blank">d69197f</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-09-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/a54969b17c22ccb84c27b57f76df3473546f29e6/docs/2025.html" target="_blank">a54969b</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/cbb6542a3df2eb9628c58499391a2f0ba2ba4b6c/docs/2025.html" target="_blank">cbb6542</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/7dba35fa6a5a0beac858da7a1b2436d1a47423ee/docs/2025.html" target="_blank">7dba35f</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9efa99a5847c25a4ed858f55d8a56f4da09c7a84/analysis/2025.Rmd" target="_blank">9efa99a</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/9efa99a5847c25a4ed858f55d8a56f4da09c7a84/docs/2025.html" target="_blank">9efa99a</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/2a565ec980344b08f7520807211647085a17df4f/docs/2025.html" target="_blank">2a565ec</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/f54695a00beb19b1e5811936abf1f4d8e7f861fc/analysis/2025.Rmd" target="_blank">f54695a</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/f54695a00beb19b1e5811936abf1f4d8e7f861fc/docs/2025.html" target="_blank">f54695a</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/2be8dd1099ad99a86d847529f7143906375044e9/docs/2025.html" target="_blank">2be8dd1</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/91c905d05f8077090ce79da3b7bb426e56b327de/analysis/2025.Rmd" target="_blank">91c905d</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/91c905d05f8077090ce79da3b7bb426e56b327de/docs/2025.html" target="_blank">91c905d</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/517d8ee0cf2f6260fababf061c4c1c3943fdedc1/docs/2025.html" target="_blank">517d8ee</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/04ff8c6e454d1b819dfba3a0d59d76bf8b2aab23/analysis/2025.Rmd" target="_blank">04ff8c6</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
test2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/264ee792cce4e152f10bbd9c8cc13fc29ec3ce9f/docs/2025.html" target="_blank">264ee79</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/6715fd7bb94d4d8e19de66c94329c2a7d2d8a9f6/analysis/2025.Rmd" target="_blank">6715fd7</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/6715fd7bb94d4d8e19de66c94329c2a7d2d8a9f6/docs/2025.html" target="_blank">6715fd7</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/e87110045e7f72b3b50d01d80a2484be5257b5c3/docs/2025.html" target="_blank">e871100</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/f587a2cc54fd9c22255c77540cfc03da3f93242f/analysis/2025.Rmd" target="_blank">f587a2c</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/f587a2cc54fd9c22255c77540cfc03da3f93242f/docs/2025.html" target="_blank">f587a2c</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/1b21f6405a44cde4e4c6aefd5db92543400eb768/docs/2025.html" target="_blank">1b21f64</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/102dcb4ab68fc3c0e4b5fe68107775b29f538988/docs/2025.html" target="_blank">102dcb4</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update:20250820
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/3dfb98d9343caa3d7fefa06f6afb63fbf049f93a/docs/2025.html" target="_blank">3dfb98d</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/c354798100cfabdb968d3b2210688ad3a96b18d2/analysis/2025.Rmd" target="_blank">c354798</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
<td>
Update_2025_08_20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/41677551966cd4045d7a8f43c1d5eb05c63dc083/docs/2025.html" target="_blank">4167755</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/de7deaccddfed74828fd8559b486411c33881da4/analysis/2025.Rmd" target="_blank">de7deac</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-07-08
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hmutanqilong/PostdocLog/681c96561551edeee7502607d9c58a17ff403170/docs/2025.html" target="_blank">681c965</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/bf2c2a4fe400fd356cc1cffefd26058f1edee2f7/analysis/2025.Rmd" target="_blank">bf2c2a4</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-07-08
</td>
<td>
Publish finished local site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
blockquote {
    padding: 10px 20px;
    margin: 0 0 20px;
    font-size: 14px;
    border-left: 5px solid #eee;
}
</style>
<div id="core-gene-discovery" class="section level2">
<h2>Core Gene Discovery</h2>
<div id="some-term-definitions-and-descriptions" class="section level3">
<h3>1. some term definitions and descriptions</h3>
<ul>
<li><p><strong>Perturbation</strong>: A change to the genome of a cell carried out by the CRISPR-Cas9 system or one of its variants. Common perturbations include CRISPRko (knockout via nuclease-active Cas9), CRISPRi (inactivation via dCas9 tethered to a repressive domain), CRISPRa (activation via dCas9 tethered to an activating domain).</p></li>
<li><p><strong>gRNA</strong>: An RNA guiding the Cas9 to its target in order to perturb it.</p></li>
<li><p><strong>Target</strong>: The genomic element targeted by a gRNA, including gene TSS, enhancer. Typically, mulitple gRNAs are designed to perturb one given target.</p></li>
<li><p><strong>Targeting gRNA</strong>: A gRNA that is intended to perturb a genomic target.</p></li>
<li><p><strong>Non-targeting gRNA</strong>: A gRNA which maps to a location whose perturbation is known to have no effect on target gene. In general, we treat Non-targeting gRNAs negative controls.</p></li>
<li><p><strong>Response</strong>: A molecular readout (gene expression) in the single-cell CRISPR screen, which is the response of perturbation of interest.</p></li>
<li><p><strong>Multiplicity of infection (MOI)</strong>: The MOI of a CRSIPR screen can be categorized as low or high and also can also be numerically quantified. A high MOI dataset is the experiment that aims to insert multiple gRNAs into each cell. A low MOI dataset aims to insert a single gRNA into each cell. The MOI reflects the average of gRNAs delivered per cell. if MOI of 10 means that on average each cell was infected by 10 gRNAs.</p></li>
</ul>
<p><strong>some descriptions of sceptre</strong>:</p>
<p>main task is “Does perturbing a target impact a response?”</p>
<ul>
<li><p><strong>Target-response pair</strong>: A pair consisting of a target and a response.For example, a target-response pair could be a specific enhancer coupled to a specific gene. Each target-response pair corresponds to a hypothesis to be tested: Does perturbing the target impact the response?</p></li>
<li><p><strong>Negative control pair</strong>: A target-response pair in which the target consits of one or more non-targeting gRNAs. Negative control pairs are used to assess the calibration of a statistical analysis method.</p></li>
<li><p><strong>Positive control pair</strong>: A target-response pair in which the target is a genomic element such as enhancer or gene known to have an effect on the response. Positive control pairs are used to assess the power of a statistical analysis method.</p></li>
<li><p><strong>Cis pair</strong>: A kind of discovery target-response pair in which the target and response are located in close proximity on the same chromosome.</p></li>
<li><p><strong>Trans pair</strong>: A a kind of discovery target-response pair for which the target and response are not necessarily located in close proximity or on the same chromosome as one another.</p></li>
</ul>
</div>
</div>
<div id="july-19" class="section level2">
<h2>July 19</h2>
<ul>
<li>what I did
<ul>
<li>exploring some mechanisms of sceptre package</li>
<li>comparing NTC guides’s results between sceptre and limma</li>
<li>run trans analysis of Replogle et al K562 KD8 genome-wide perturb-seq (gwps) dataset</li>
</ul></li>
</ul>
<div id="some-mechanisms-of-sceptre-package" class="section level3">
<h3>1. some mechanisms of sceptre package</h3>
<div id="construct-cis-grna_target-response-pairs" class="section level4">
<h4><strong>1.1 construct cis gRNA_target-response pairs:</strong></h4>
<p>Why dose sceptre exclude all pairs containing each positive control gRNA target from the cis pairs when using <code>construct_cis_pairs(sceptre_object, positive_control_pairs = pc_pairs, distance_threshold = 5e6)</code>?</p>
<blockquote>
<p>sceptre authors’ thought: a gRNA target used for a positive control pair is a known biological effector. Its purpose is to produce a large and predictable effect. Including this gRNA target’s pairs in the discovery analysis, such as gRNA-other gene (A, B) pairs, violates the purpose of the discovery analysis. The analysis of these additional pairs is no longer a pure exploration of novel cis-regulatory relationships, but an analysis of the pleiotropic or off-target effects of a known, powerfull perturbation.</p>
</blockquote>
<blockquote>
<p>In statistics, <strong>1) contaminating the Null Distribution</strong>: If a positive control gRNA target is included in the analysis, its strong effect on its target gene (and potentially other genes) will compromise the whole resampleing process. The null distribution will be generated by resampling a powerfull biological effector, which can make the variance of null distribution inflated. The simulation of the null hypothesis would be contaminated by a non-null reality. <strong>2) Distorting the FDR</strong>: FDR depends on the overall p-value distribution of test. if we include a set of known, extremely significant p-values from positive control targets (whose p-values may be close to zero) in this distribution will be distorted.</p>
</blockquote>
</div>
<div id="calibration-mechanism" class="section level4">
<h4><strong>1.2 calibration mechanism</strong></h4>
<p>In the calibration step, sceptre samples groups of non-targeting gRNAs to calibrate. The idea is to get enough NTC gRNA target-response pairs to check whether the test is well calibrated. It also depends on the number of qc passed discovery pairs (here is cis pairs). It follows this equation:</p>
<p><span class="math display">\[
\text{n_NTC_gRNAs}= 
  \frac{ \text{MULT_FACTOR} \times \text{n_calibration_pairs} }
       { \text{p_hat} \times \text{n_genes} }
\]</span></p>
<p>We set MULT_FACTOR to 5 as a redundancy factor.</p>
<p>p_hat represents the estimated probability that an NTC gRNA–gene pair is valid, which is 0.96 in our case.</p>
<p>The total number of genes (n_genes) is 7,991.</p>
<p>For n_calibration_pairs, we ranged it from 101,920 (the real number of cis pairs) up to 1,000,000 so that all NTC gRNAs could be covered.</p>
<p>code below is my simulation result:</p>
<pre class="r"><code>suppressPackageStartupMessages(library(tidyverse))</code></pre>
<pre><code>unable to deduce timezone name from &#39;America/Chicago&#39;</code></pre>
<pre class="r"><code>suppressPackageStartupMessages(library(scales))

compute_n_grna_groups &lt;- function(n_calibration_pairs,
                                  n_genes = 7991,
                                  p_hat = 0.96,
                                  N_POSSIBLE_GROUPS_THRESHOLD = 100L,
                                  n_possible_groups = 514L,
                                  MULT_FACTOR = 5) {
  raw &lt;- ceiling(MULT_FACTOR * n_calibration_pairs / (p_hat * n_genes))
  lower &lt;- pmax(raw, N_POSSIBLE_GROUPS_THRESHOLD)
  out &lt;- ifelse(lower &gt;= n_possible_groups, n_possible_groups, lower)
  as.integer(out)
}

p_hat &lt;- 0.96
n_genes &lt;- 7991
N_POSSIBLE_GROUPS_THRESHOLD &lt;- 100L
n_possible_groups &lt;- 514L
MULT_FACTOR &lt;- 5

lower &lt;- N_POSSIBLE_GROUPS_THRESHOLD * p_hat * n_genes / MULT_FACTOR
upper &lt;- n_possible_groups * p_hat * n_genes / MULT_FACTOR

x_vals &lt;- seq(101920, 1100000, by = 5000)

dat &lt;- data.frame(n_calibration_pairs = x_vals)
dat$n_grna_groups &lt;- compute_n_grna_groups(
  dat$n_calibration_pairs,
  n_genes = n_genes,
  p_hat = p_hat,
  N_POSSIBLE_GROUPS_THRESHOLD = N_POSSIBLE_GROUPS_THRESHOLD,
  n_possible_groups = n_possible_groups,
  MULT_FACTOR = MULT_FACTOR
)

p &lt;- ggplot(dat, aes(x = n_calibration_pairs, y = n_grna_groups)) +
  geom_line(linewidth = 1) +
  geom_hline(yintercept = N_POSSIBLE_GROUPS_THRESHOLD, linetype = 2) + # 上下限
  geom_hline(yintercept = n_possible_groups, linetype = 2) +
  geom_vline(xintercept = lower, linetype = 3) +
  geom_vline(xintercept = upper, linetype = 3) +
  annotate(&quot;text&quot;, x = lower, y = N_POSSIBLE_GROUPS_THRESHOLD + 20,
           label = sprintf(&quot;lower ≈ %.0f&quot;, lower), angle = 90, vjust = -0.5, hjust = 0) +
  annotate(&quot;text&quot;, x = upper, y = n_possible_groups - 20,
           label = sprintf(&quot;upper ≈ %.0f&quot;, upper), angle = 90, vjust = 1.1, hjust = 1) +
  labs(
    x = &quot;n_calibration_pairs&quot;,
    y = &quot;n_grna_groups&quot;,
    title = &quot;p_hat=0.96, n_genes=7991&quot;,
  ) +
  scale_y_continuous(breaks = c(100, 200, 300, 400, 514)) +
  scale_x_continuous(labels = label_number()) +
  theme_classic(base_size = 13)

print(p)</code></pre>
<p><img src="figure/2025.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/3dfb98d9343caa3d7fefa06f6afb63fbf049f93a/docs/figure/2025.Rmd/unnamed-chunk-1-1.png" target="_blank">3dfb98d</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="comparison-of-ntc-grna-alibration-between-sceptre-and-limma" class="section level3">
<h3>2. Comparison of NTC gRNA alibration between sceptre and limma</h3>
<p>We compared calibration results of non-targeting gRNAs between limma and sceptre by aligning shared ntc gRNA–gene pairs (~1M) and visualizing overlap with Venn diagrams, QQ plots, and scatter plots to assess concordance and differences in significance.</p>
<p>In sceptre analysis, we manually set <code>n_calibration_pairs = 1000000</code>, which included all 514 NTC gRNAs in the calibration. On average, each NTC gRNA was paired with 1,946 genes as target–response pairs. We totally found 24 significant ntc gRNA pairs at alpha=0.1.</p>
<p>In the limma analysis, every NTC gRNA was paired with the entire set of genes without any restriction. So we obtained 1000,000 shared ntc gRNA pairs.</p>
<pre class="r"><code>knitr::include_graphics(&quot;assets/2025/figure/limma_sceptre_shared_pairs.png&quot;, error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<img src="assets/2025/figure/limma_sceptre_shared_pairs.png" alt="venn diagram of shared ntc gRNA pairs" width="50%" />
<p class="caption">
venn diagram of shared ntc gRNA pairs
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-limma_sceptre_shared_pairs">
Past versions of limma_sceptre_shared_pairs.png
</button>
</p>
<div id="fig-limma_sceptre_shared_pairs" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9707b9e08a5c8aad5a1ae0207983993fe4362b7f/docs/assets/2025/figure/limma_sceptre_shared_pairs.png" target="_blank">9707b9e</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>knitr::include_graphics(
  c(&quot;assets/2025/figure/limma_all_pairs_qqplot.png&quot;, 
    &quot;assets/2025/figure/sceptre_all_pairs_qqplot.png&quot;), error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<p><img src="assets/2025/figure/limma_all_pairs_qqplot.png" alt="QQ plot stratified by ntc gRNAs" width="49%" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-limma_all_pairs_qqplot">
Past versions of limma_all_pairs_qqplot.png
</button>
</p>
<div id="fig-limma_all_pairs_qqplot" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9707b9e08a5c8aad5a1ae0207983993fe4362b7f/docs/assets/2025/figure/limma_all_pairs_qqplot.png" target="_blank">9707b9e</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<img src="assets/2025/figure/sceptre_all_pairs_qqplot.png" alt="QQ plot stratified by ntc gRNAs" width="49%" />
<p class="caption">
QQ plot stratified by ntc gRNAs
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sceptre_all_pairs_qqplot">
Past versions of sceptre_all_pairs_qqplot.png
</button>
</p>
<div id="fig-sceptre_all_pairs_qqplot" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9707b9e08a5c8aad5a1ae0207983993fe4362b7f/docs/assets/2025/figure/sceptre_all_pairs_qqplot.png" target="_blank">9707b9e</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>knitr::include_graphics(
  c(&quot;assets/2025/figure/limma_vs_sceptre_shared_pairs_qqplot.png&quot;,
    &quot;assets/2025/figure/limma_vs_sceptre_shared_pairs_scatter_plot.png&quot;), error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<p><img src="assets/2025/figure/limma_vs_sceptre_shared_pairs_qqplot.png" alt="shared pairs QQ plot and scatter plot: limma v.s sceptre" width="49%" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-limma_vs_sceptre_shared_pairs_qqplot">
Past versions of limma_vs_sceptre_shared_pairs_qqplot.png
</button>
</p>
<div id="fig-limma_vs_sceptre_shared_pairs_qqplot" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9707b9e08a5c8aad5a1ae0207983993fe4362b7f/docs/assets/2025/figure/limma_vs_sceptre_shared_pairs_qqplot.png" target="_blank">9707b9e</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<img src="assets/2025/figure/limma_vs_sceptre_shared_pairs_scatter_plot.png" alt="shared pairs QQ plot and scatter plot: limma v.s sceptre" width="49%" />
<p class="caption">
shared pairs QQ plot and scatter plot: limma v.s sceptre
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-limma_vs_sceptre_shared_pairs_scatter_plot">
Past versions of limma_vs_sceptre_shared_pairs_scatter_plot.png
</button>
</p>
<div id="fig-limma_vs_sceptre_shared_pairs_scatter_plot" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hmutanqilong/PostdocLog/blob/9707b9e08a5c8aad5a1ae0207983993fe4362b7f/docs/assets/2025/figure/limma_vs_sceptre_shared_pairs_scatter_plot.png" target="_blank">9707b9e</a>
</td>
<td>
hmutanqilong
</td>
<td>
2025-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="simulation-of-batch-as-continuous-or-categorial-varibale" class="section level3">
<h3>3. simulation of batch as continuous or categorial varibale</h3>
<p><strong>Design</strong>:</p>
<p>We simulated a null setting where gene expression does not depend on batch. We generated counts for 300 batches × 20 cells per batch from Poisson(μ = 10). For each replicate (500 in total) we fit three glm regressons:</p>
<ol style="list-style-type: decimal">
<li><p>y ~ 1 (intercept-only, the true model)</p></li>
<li><p>y ~ batch with batch as continuous</p></li>
<li><p>y ~ factor(batch) with batch as categorical</p></li>
</ol>
<p>From each Poisson fit we took the fitted means and then estimated the NB size parameter . I calculated residual deviance/df, , and RMSE between and true .</p>
<p><strong>Results</strong>: Across 50 replicates, all three models had deviance/df = 1 and very large , suggesting that near-Poisson residuals and no material overdispersion. the p value of in numeric model was with an approximately uniform p-value distribution. For mean estimation, the value of intercept-only is similar with the value of numeric model (small RMSE), but factor had a much larger RMSE. I guess it fits a separate mean per batch from small per-batch sample sizes (only 20 cells).</p>
<pre class="r"><code>set.seed(1)
NSIM  &lt;- 500
N_BATCH &lt;- 300
CELLS_PER_BATCH &lt;- 20 #每批细胞数量(均衡)
MU_TRUE &lt;- 10 # 各批次真实均值（完全相同）
THETA_CAP &lt;- 1000 # theta上限 sceptre setting is 1000
THETA_LO  &lt;- 1e-2 # theta下限


library(MASS) # theta.md</code></pre>
<pre><code>
Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre class="r"><code>library(tidyverse)
library(patchwork)</code></pre>
<pre><code>
Attaching package: &#39;patchwork&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MASS&#39;:

    area</code></pre>
<pre class="r"><code>library(scales)

# helper function to estimate theta
theta_est &lt;- function(y, mu, dfr, cap = THETA_CAP, lo = THETA_LO) {
  # th_list &lt;- sceptre:::estimate_theta(
  #   y = y, mu = mu, dfr = dfr,
  #   limit = 50, eps = (.Machine$double.eps)^(1/4)
  # )
  # th &lt;- th_list[[1]]
  # th &lt;- max(min(th, cap), lo)
  # th
  
  th &lt;- tryCatch(MASS::theta.md(y, mu, dfr), error = function(e) NA_real_)
  th &lt;- max(min(th, cap), lo)
  th
}

# one time simulation
# real: 各批次间无差异，即同均值。
# aim: 此时若把batch加入到协变量中作为连续性变量，是否会使poisson 估计的mu有偏？

fit_once_equal_mu &lt;- function(n_batch = N_BATCH, cells_per_batch = CELLS_PER_BATCH, mu_true = MU_TRUE) {
  # 构造均衡批次
  batch &lt;- rep(seq_len(n_batch), each = cells_per_batch)
  
  # 真实情况：所有批次均值完全相同
  y &lt;- rpois(length(batch), lambda = mu_true)
  dat &lt;- data.frame(y = y, batch = batch)
  
  # A) 真模型：截距-only
  m_int  &lt;- glm(y ~ 1, data = dat, family = poisson())
  mu_int &lt;- fitted(m_int)
  th_int &lt;- theta_est(dat$y, mu_int, df.residual(m_int))
  out_int &lt;- tibble(
    model   = &quot;Intercept-only (true)&quot;,
    theta   = th_int,
    dev     = deviance(m_int),
    df      = df.residual(m_int),
    dev_df  = dev / df,
    beta    = NA_real_, se_beta = NA_real_, z_beta = NA_real_, p_beta = NA_real_,
    RMSE_mu = sqrt(mean((mu_int - mu_true)^2))
  )
  
  # B) 把 batch 当作连续
  m_num  &lt;- glm(y ~ batch, data = dat, family = poisson())
  mu_num &lt;- fitted(m_num)
  th_num &lt;- theta_est(dat$y, mu_num, df.residual(m_num))
  co &lt;- coef(summary(m_num))
  beta_hat &lt;- unname(co[2, &quot;Estimate&quot;])
  se_hat &lt;- unname(co[2, &quot;Std. Error&quot;])
  z_hat &lt;- unname(co[2, &quot;z value&quot;])
  p_hat &lt;- unname(co[2, &quot;Pr(&gt;|z|)&quot;])
  out_num &lt;- tibble(
    model   = &quot;Numeric (batch as continuous)&quot;,
    theta   = th_num,
    dev     = deviance(m_num),
    df      = df.residual(m_num),
    dev_df  = dev / df,
    beta    = beta_hat, se_beta = se_hat, z_beta = z_hat, p_beta = p_hat,
    RMSE_mu = sqrt(mean((mu_num - mu_true)^2))
  )
  
  # C) 把 batch 当作因子 or dummy matrix
  m_fac  &lt;- glm(y ~ factor(batch), data = dat, family = poisson())
  mu_fac &lt;- fitted(m_fac)
  th_fac &lt;- theta_est(dat$y, mu_fac, df.residual(m_fac))
  out_fac &lt;- tibble(
    model = &quot;Factor (batch as categorical)&quot;,
    theta = th_fac,
    dev = deviance(m_fac),
    df = df.residual(m_fac),
    dev_df = dev / df,
    beta = NA_real_, se_beta = NA_real_, z_beta = NA_real_, p_beta = NA_real_,
    RMSE_mu = sqrt(mean((mu_fac - mu_true)^2))
  )
  
  bind_rows(out_int, out_num, out_fac)
}

NSIM &lt;- 50
res &lt;- bind_rows(replicate(NSIM, fit_once_equal_mu(), simplify = FALSE)) %&gt;%
  group_by(model) %&gt;% mutate(sim = row_number()) %&gt;% ungroup()

summ &lt;- res %&gt;%
  group_by(model) %&gt;%
  summarise(
    SIM_n = n(),
    mean_dev_df = mean(dev_df, na.rm = TRUE),
    sd_dev_df = sd(dev_df,   na.rm = TRUE),
    median_theta = median(theta, na.rm = TRUE),
    prop_theta_cap = mean(theta &gt;= THETA_CAP, na.rm = TRUE),
    mean_RMSE_mu = mean(RMSE_mu, na.rm = TRUE),
    .groups = &quot;drop&quot;
  )

kableExtra::kable(summ)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">SIM_n</th>
<th align="right">mean_dev_df</th>
<th align="right">sd_dev_df</th>
<th align="right">median_theta</th>
<th align="right">prop_theta_cap</th>
<th align="right">mean_RMSE_mu</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Factor (batch as categorical)</td>
<td align="right">50</td>
<td align="right">1.019100</td>
<td align="right">0.0176873</td>
<td align="right">430.5795</td>
<td align="right">0.24</td>
<td align="right">0.7100287</td>
</tr>
<tr class="even">
<td align="left">Intercept-only (true)</td>
<td align="right">50</td>
<td align="right">1.018741</td>
<td align="right">0.0173974</td>
<td align="right">423.0253</td>
<td align="right">0.24</td>
<td align="right">0.0319833</td>
</tr>
<tr class="odd">
<td align="left">Numeric (batch as continuous)</td>
<td align="right">50</td>
<td align="right">1.018792</td>
<td align="right">0.0174404</td>
<td align="right">436.1709</td>
<td align="right">0.26</td>
<td align="right">0.0480753</td>
</tr>
</tbody>
</table>
</div>
<div id="trans-analysis" class="section level3">
<h3>trans analysis</h3>
<pre class="r"><code>library(arrow)</code></pre>
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
<pre><code>
Attaching package: &#39;arrow&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    timestamp</code></pre>
<pre class="r"><code>library(duckplyr)</code></pre>
<pre><code>The duckplyr package is configured to fall back to dplyr when it encounters an
incompatibility. Fallback events can be collected and uploaded for analysis to
guide future development. By default, data will be collected but no data will
be uploaded.
ℹ Automatic fallback uploading is not controlled and therefore disabled, see
  `?duckplyr::fallback()`.
✔ Number of reports ready for upload: 26373.
→ Review with `duckplyr::fallback_review()`, upload with
  `duckplyr::fallback_upload()`.
ℹ Configure automatic uploading with `duckplyr::fallback_config()`.</code></pre>
<pre><code>✔ Overwriting dplyr methods with duckplyr methods.
ℹ Turn off with `duckplyr::methods_restore()`.</code></pre>
<pre class="r"><code># db &lt;- open_dataset(&#39;/project/xuanyao/qilong/Project/Core_Gene_Programs/Johnathan/sceptre/K562_gwps_all_in_one/sceptre_outputs_trans/trans_results/&#39;)
# write_parquet(db, &#39;/project/xuanyao/qilong/Project/Core_Gene_Programs/Johnathan/sceptre/K562_gwps_all_in_one/sceptre_outputs_trans/Reploge_2022_Cell_K562_gwps_KD8_all_trans_results.parquet&#39;)

db &lt;- read_parquet_duckdb(&#39;/project/xuanyao/qilong/Project/Core_Gene_Programs/Johnathan/sceptre/K562_gwps_all_in_one/sceptre_outputs_trans/Reploge_2022_Cell_K562_gwps_KD8_all_trans_results.parquet&#39;)

tmp &lt;- db %&gt;% filter(response_id==&#39;ENSG00000206172&#39; &amp; pass_qc == T &amp; grna_target == &#39;ENSG00000105610&#39;) 

kableExtra::kable(tmp)</code></pre>
<table>
<colgroup>
<col width="13%" />
<col width="17%" />
<col width="13%" />
<col width="11%" />
<col width="13%" />
<col width="6%" />
<col width="8%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">response_id</th>
<th align="left">grna_id</th>
<th align="left">grna_target</th>
<th align="right">n_nonzero_trt</th>
<th align="right">n_nonzero_cntrl</th>
<th align="left">pass_qc</th>
<th align="right">p_value</th>
<th align="right">log_2_fold_change</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000206172</td>
<td align="left">gRNA:ENSG00000105610</td>
<td align="left">ENSG00000105610</td>
<td align="right">22</td>
<td align="right">18518</td>
<td align="left">TRUE</td>
<td align="right">0.0008608</td>
<td align="right">-0.9257564</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="oct-22" class="section level2">
<h2>Oct 22</h2>
<div id="collected-and-compiled-public-available-perturb-seq-datasets" class="section level3">
<h3>1 Collected and compiled public available Perturb-seq datasets</h3>
<p>I updated the Perturb-seq data summary document <a href="https://docs.google.com/spreadsheets/d/1eXf6x3jSWxG3dgwhATYLRH8pcKLtk4-tJ3w-1xlDUzY/edit?usp=sharing">link</a>, and provided some phenotype information which these datasets could be applied to.</p>
</div>
<div id="pilot-cell-line-and-disease" class="section level3">
<h3>2 Pilot cell line and disease</h3>
<p>We firstly select <strong>age-related macular degeneration (AMD) disease</strong>, an eye’s disorders, as a pilot trait, because it is considered a well-established case in GWAS studies and has been identified some important core genes including CFH, ARMS2, and HTRA1.</p>
<p>The related perturbation data is based on RPE1 (retinal pigment epithelial) cell line, which was reported by Replogle et al., 2022, Cell.</p>
<p>Here are some descriptions of this dataset.</p>
<p>The RPE1 data in their paper was just perturbed a set of essential genes using CRISPRi tech. These essential genes include :</p>
<ol style="list-style-type: decimal">
<li><p>20Q1 Cancer Dependency Map common essential genes (<a href="https://depmap.org/portal/download/" class="uri">https://depmap.org/portal/download/</a>);</p></li>
<li><p>a number of hand-selected genes with interesting phenotypes in the K562 genome-wide Perturb-seq dataset;</p></li>
<li><p>non-targeting control sgRNAs accounting for 5% of the total library.</p></li>
</ol>
<p>The Venn diagram shows that a few AMD-related candidate genes overlap.</p>
<pre class="r"><code>knitr::include_graphics(&quot;assets/2025/figure/Oct/venn_amd_rpe1.png&quot;, error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<img src="assets/2025/figure/Oct/venn_amd_rpe1.png" alt="overlapped AMD-related genes from GWAScatalog" width="50%" />
<p class="caption">
overlapped AMD-related genes from GWAScatalog
</p>
</div>
</div>
<div id="perturb-seq-single-cell-data-process" class="section level3">
<h3>3. Perturb-seq single cell data process</h3>
<p>Perturb-seq data was download from figshare. File name is rpe1_raw_singlecell_01.h5ad, which means raw, single-cell expression data for genes expressed at &gt;0.01 UMI per cell.</p>
<p>In our HPC, I performed such quality control steps:</p>
<ol start="0" style="list-style-type: decimal">
<li><p>raw sc data <strong>[n_obs × n_vars = 247914 × 8749]</strong></p></li>
<li><p>remove the cell without sgRNA <strong>[n_obs × n_vars = 247456 × 8749] (-458 cells)</strong></p></li>
<li><p>check sgRNA_id mismatch target gene <strong>[n_obs × n_vars = 247456 × 8749] (PASS)</strong></p></li>
<li><p>remove sgRNA carried by less than 20 cells <strong>[n_obs × n_vars = 244984 × 8749] (-189 cells)</strong></p></li>
<li><p>tag MT-gene, conduct qc metrics</p></li>
<li><p>remove cells with mito percent &gt; 20% <strong>[n_obs × n_vars = 244984 × 8749] (0 cells)</strong></p></li>
<li><p>remove genes at chrX/Y, chrM <strong>[n_obs × n_vars = 244984 × 8469] (-280 genes)</strong></p></li>
<li><p>remove outlier cells <strong>[n_obs × n_vars = 244493 × 8469] (-491 cells)</strong></p></li>
</ol>
<p>finally: n_obs × n_vars = 244493 × 8469; num sgRNA: 2201 target sgRNA + 1 non-targeting; non-targeting cells: 11473</p>
</div>
<div id="burden-test-association-of-amd-disease" class="section level3">
<h3>4. Burden test association of AMD disease</h3>
<p>Summary statistic was obtained from the UK Biobank, reported by Backman et al., Nature (2021). Data was downloaded from GWAS Catalog (GCST90083420).</p>
<p>They provided odds ratios, so I converted them to beta using log(OR). I then selected the results that only include pLOF variants (MAF &lt; 0.01), corresponding to the M.001 flag.</p>
<p>The P value distribution is that:</p>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(tidyverse)
library(patchwork)

burden_eff &lt;- fread(&#39;/project/xuanyao/qilong/Project/Core_Gene_Programs_II/03_burden_test/macular_degeneration/processed_data/LOF_burdenTest_M1.001_anno.tsv&#39;)
burden_eff &lt;- burden_eff %&gt;% 
  mutate(minlogp = -log10(p_value),
         label_gene = gene_name)

burden_eff$label_gene[which(burden_eff$p_value &gt; 0.001)] &lt;- NA

p1_hist &lt;- ggplot(data = burden_eff, aes(x = p_value)) + 
  geom_histogram(color = &#39;black&#39;, fill = &#39;royalblue&#39;, bins = 50) +
  ggtitle(&quot;total genes: 14,261&quot;) +
  theme_classic() +
  theme(title = element_text(size = 10), 
        axis.title = element_text(size = 12))

p2_hist &lt;- ggplot(data = burden_eff[burden_eff$p_value &lt; 0.01, ], aes(x = p_value)) + 
  geom_histogram(color = &#39;black&#39;, fill = &#39;royalblue&#39;) +
  ggtitle(&quot;total genes: 120 (p value &lt; 0.01)&quot;) +
  theme_classic() +
  theme(title = element_text(size = 10), 
        axis.title = element_text(size = 12))


p3_valcano &lt;- ggplot(burden_eff, aes(x = beta, y = minlogp, label = label_gene)) + 
  geom_point(size = 0.7) + 
  ggrepel::geom_text_repel(size = 3) +
  theme_classic()
  
p1_hist + p2_hist + p3_valcano + plot_layout(ncol = 2, nrow = 2, byrow = T)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/2025.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="trans-target-gene-pairs-association-using-sceptre" class="section level3">
<h3>5. trans target-gene pairs association using SCEPTRE</h3>
<p>I used all trans pairs to perform the calibration step, but the results still showed some inflation. I then plotted a stratified QQ plot by sgRNA.</p>
<p>A total of 264 pairs showed a difference greater than 1.0 between the observed and expected P-values.</p>
<pre class="r"><code>knitr::include_graphics(&quot;assets/2025/figure/Oct/plot_stratified_qqplot.png&quot;, error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<img src="assets/2025/figure/Oct/plot_stratified_qqplot.png" alt="non-targeting sgRNA p value distribution" width="50%" />
<p class="caption">
non-targeting sgRNA p value distribution
</p>
</div>
<p>And sceptre p value distribution is that:</p>
<pre class="r"><code>knitr::include_graphics(&quot;assets/2025/figure/Oct/sceptre_rpe1_p_distribution.png&quot;, error = FALSE)</code></pre>
<div class="figure" style="text-align: center">
<img src="assets/2025/figure/Oct/sceptre_rpe1_p_distribution.png" alt="SCEPTRE trans discovery pairs p value distribution" width="70%" />
<p class="caption">
SCEPTRE trans discovery pairs p value distribution
</p>
</div>
</div>
<div id="core-gene-identification-using-dandelion-and-otas-causal-model" class="section level3">
<h3>6. Core gene identification using Dandelion and Ota’s causal model</h3>
<p>Both two methods didn’t identify any signals. Possible reasons might include unmatched perturbed genes corresponding to the AMD trait and the much lower number of significant genes in the burden test (there is only one gene which passed the genome-wide threshold).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 8

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el8-x86_64/lib/libopenblas_skylakexp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.8    duckplyr_1.1.0       arrow_20.0.0.2      
 [4] patchwork_1.3.1.9000 MASS_7.3-60          scales_1.3.0        
 [7] forcats_0.5.1        stringr_1.5.1        dplyr_1.1.4         
[10] purrr_1.0.2          readr_1.4.0          tidyr_1.3.0         
[13] tibble_3.2.1         ggplot2_3.5.1        tidyverse_1.3.1     
[16] workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.9        bit64_4.6.0-1     jsonlite_1.8.7   
 [5] viridisLite_0.4.2 modelr_0.1.8      bslib_0.7.0       assertthat_0.2.1 
 [9] getPass_0.2-2     highr_0.11        cellranger_1.1.0  ggrepel_0.9.6    
[13] yaml_2.2.1        collections_0.3.8 pillar_1.10.2     backports_1.4.1  
[17] glue_1.6.2        digest_0.6.33     promises_1.2.0.1  rvest_1.0.0      
[21] colorspace_2.1-0  htmltools_0.5.8.1 httpuv_1.6.1      pkgconfig_2.0.3  
[25] broom_0.7.8       haven_2.4.1       processx_3.8.4    svglite_2.0.0    
[29] whisker_0.4.1     later_1.2.0       git2r_0.32.0      generics_0.1.3   
[33] farver_2.1.1      ellipsis_0.3.2    cachem_1.0.8      withr_2.5.2      
[37] cli_3.6.1         magrittr_2.0.3    crayon_1.5.2      readxl_1.3.1     
[41] memoise_2.0.1     evaluate_0.23     ps_1.7.5          fs_1.6.3         
[45] xml2_1.3.2        tools_4.1.0       hms_1.1.0         lifecycle_1.0.4  
[49] munsell_0.5.0     reprex_2.0.0      callr_3.7.6       kableExtra_1.4.0 
[53] compiler_4.1.0    jquerylib_0.1.4   duckdb_1.3.1      systemfonts_1.1.0
[57] rlang_1.1.6       grid_4.1.0        rstudioapi_0.15.0 labeling_0.4.3   
[61] rmarkdown_2.27    gtable_0.3.6      DBI_1.2.3         R6_2.5.1         
[65] lubridate_1.7.10  knitr_1.48        fastmap_1.1.1     bit_4.0.5        
[69] rprojroot_2.0.4   stringi_1.6.2     Rcpp_1.0.12       vctrs_0.6.4      
[73] dbplyr_2.1.1      tidyselect_1.2.0  xfun_0.45        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "site_libs/mathjax-local/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
